packages/pytest-simcore/src/pytest_simcore/playwright_utils.py

```diff
@@ -67,6 +72,24 @@ def retrieve_project_state_from_decoded_message(event: SocketIOEvent) -> Running
     return RunningState(event.obj["data"]["state"]["value"])
 
 
+@dataclass
+class SocketIOProjectClosedWaiter:
+    def __call__(self, message: str) -> bool:
+        with log_context(logging.DEBUG, msg=f"handling websocket {message=}"):
+            # socket.io encodes messages like so
+            # https://stackoverflow.com/questions/24564877/what-do-these-numbers-mean-in-socket-io-payload
+            if message.startswith("42"):
+                decoded_message = decode_socketio_42_message(message)
+                if (
+                    (decoded_message.name == "projectStateUpdated")
```

Comment:
Q: how do we know this is in sync with the code-base? e.g. assume we change `projectStateUpdated` but `ProjectStateUpdated`

Reply:
Well the e2e test in the ci will fail

Reply:
and I'm now using some constants